pkg/reconciler/managed/reconciler.go

```diff
@@ -83,6 +87,46 @@ const (
 	reasonReconciliationPaused event.Reason = "ReconciliationPaused"
 )
 
+var supportedManagementPolicies = []sets.Set[xpv1.ManagementAction]{
+	// Default (all), the standard behaviour of crossplane in which all
+	// reconciler actions are done.
+	sets.New[xpv1.ManagementAction](xpv1.ManagementActionAll),
+	// All actions explicitly set, the same as default.
+	sets.New[xpv1.ManagementAction](xpv1.ManagementActionObserve, xpv1.ManagementActionCreate, xpv1.ManagementActionUpdate, xpv1.ManagementActionLateInitialize, xpv1.ManagementActionDelete),
+	// ObserveOnly, just observe action is done, the external resource is
+	// considered as read-only.
+	sets.New[xpv1.ManagementAction](xpv1.ManagementActionObserve),
+	// Pause, no action is being done. Alternative to setting the pause
+	// annotation.
+	sets.New[xpv1.ManagementAction](),
+	// No LateInitialize filling in the spec.forProvider, allowing some
+	// external resource fields to be managed externally.
+	sets.New[xpv1.ManagementAction](xpv1.ManagementActionObserve, xpv1.ManagementActionCreate, xpv1.ManagementActionUpdate, xpv1.ManagementActionDelete),
+	// No Delete, the external resource is not deleted when the managed
+	// resource is deleted.
+	sets.New[xpv1.ManagementAction](xpv1.ManagementActionObserve, xpv1.ManagementActionCreate, xpv1.ManagementActionUpdate, xpv1.ManagementActionLateInitialize),
+	// No Delete and no LateInitialize, the external resource is not deleted
+	// when the managed resource is deleted and the spec.forProvider is not
+	// late initialized.
+	sets.New[xpv1.ManagementAction](xpv1.ManagementActionObserve, xpv1.ManagementActionCreate, xpv1.ManagementActionUpdate),
+	// No Update, the external resource is not updated when the managed
+	// resource is updated. Useful for immutable external resources.
+	sets.New[xpv1.ManagementAction](xpv1.ManagementActionObserve, xpv1.ManagementActionCreate, xpv1.ManagementActionDelete, xpv1.ManagementActionLateInitialize),
+	// No Update and no Delete, the external resource is not updated
+	// when the managed resource is updated and the external resource
+	// is not deleted when the managed resource is deleted.
+	sets.New[xpv1.ManagementAction](xpv1.ManagementActionObserve, xpv1.ManagementActionCreate, xpv1.ManagementActionLateInitialize),
+	// No Update and no LateInitialize, the external resource is not updated
+	// when the managed resource is updated and the spec.forProvider is not
+	// late initialized.
+	sets.New[xpv1.ManagementAction](xpv1.ManagementActionObserve, xpv1.ManagementActionCreate, xpv1.ManagementActionDelete),
+	// No Update, no Delete and no LateInitialize, the external resource is
+	// not updated when the managed resource is updated, the external resource
+	// is not deleted when the managed resource is deleted and the
+	// spec.forProvider is not late initialized.
+	sets.New[xpv1.ManagementAction](xpv1.ManagementActionObserve, xpv1.ManagementActionCreate),
```

Comment:
Should we also add Observe + Update ? https://github.com/crossplane-contrib/provider-kubernetes/issues/115

Reply:
Nice, thanks for updating these comments

I'd prefer to avoid this giant set (no pun intended) of package scoped state. Perhaps we could make it a member of `ManagementPolicyActionChecker` and move its declaration into the  `NewManagementPolicyActionChecker`?

Reply:
moved the giant set into an func returning default supported policies and made it injectable for the `ManagedPoliciesResolver`, so that its not package-scoped anymore and we can still easily change it for tests.

Reply:
So, we don't want to add `Observe` + `Update` for now but enable a way for individual providers to extend the supported list? When I check the code, this doesn't seem to be the case though.

Reply:
Actually I want to add `Observe + Update` but didnt yet have a chance to add and test it first. Actually maybe it would be better to add them in a next PR, so that this can be merged?

I havent thought about allowing providers to set their own supported policies, now they are injected just in the resovler which actually just helps with tests and not having the supported list package scoped. But I like the idea, we can inject the list into the Reconciler, so that the providers can change it easier if needed

Reply:
@turkenh now the supported policies can be injected into the Reconciler, so if some provider devs would like to test something new or have their own set, its possible :tada: